### PR TITLE
Added support for native API for Verizon SE 1.2.1

### PR DESCRIPTION
**Pull request by lcsabaignat**
Adding support for Rewarded Video
Adding support for Native

### CODE REVIEW

**Comment by maninomula** [comment]
I have made a commit to this PR. So overdid the CLA to Yes.

**Comment by maninomula** [comment]
@lcsabaignat , Thanks for the PR. We had made some changes to the PR, please take a look in to the changes and get back here if you have any questions. The main changes we did here are
1) Removed `GADRewardBasedVideoAdConnector` as it is no longer needed, used `loadRewardedAdForAdConfiguration:completionHandler:` instead.
2) Modified `GADMAdapterVerizonNativeAd` and `GADMAdapterVerizonRewardedAd` classes to request and delegate callbacks from Verizon Media SDK.
3) Added `setUpWithConfiguration:completionHandler:` in `GADMediationAdapterVerizon` class which initializes Verizon SDK as the app init time rather than at ad request time.
4) Changed the code to match Google's objective-c code style.

File: adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+    
+    self.vasAds.requestMetadata = [builder build];
+}
+
+- (void)setCoppaFromConnector:(id<GADMediationAdRequest>)connector
+{
+    self.vasAds.COPPA = [connector childDirectedTreatment];
+}
+
+- (CGRect)makeBannerRect:(GADAdSize)adSize
```
Comment (maninomula) [inline adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m]: Why did we remove the adaptive banner code and added `makeBannerRect: ` method?

File: adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+{
+    // Consideration for smart banners ( http://bit.ly/1wAWn0r )
+    CGSize windowSize = [[UIApplication sharedApplication] keyWindow].bounds.size;
+    
+    if ( GADAdSizeEqualToSize(kGADAdSizeSmartBannerPortrait, adSize) ||
```
Comment (maninomula) [inline adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m]: This condition always fails.

**Comment by lcsabaignat** [comment]
@maninomula Thank you for your review. We are looking into it, making changes, and we will get back to you ASAP.

**Comment by maninomula** [comment]
I have made some changes to the PR so added CLA as yes.

File: adapters/VerizonMedia/VerizonMediaAdapter/GADMAdapterNexage.m [hunk 3]
```diff
@@ -5,56 +5,42 @@
...
-{
-    if (self = [super initWithRewardBasedVideoAdNetworkConnector:rewardConnector]) {
-        [self initializeVASAds];
-    }
-    return self;
-}
+- (void)initializeVASAds {
+  // Position.
+  NSDictionary *credentials = [self.connector credentials];
+  if (credentials[kGADMAdapterVerizonMediaPosition]) {
```
[the remark this answers is not in the document]
Comment (maninomula) [reply adapters/VerizonMedia/VerizonMediaAdapter/GADMAdapterNexage.m]: This change is not intended. Reverted.

File: adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m (hunk 1 above)
Comment (maninomula) [reply adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m]: I believe this is expected behavior as Google Mobile Ads SDK checks if the available sizes satisfies the min fill ratio required to accept the size. The supported size width should fill more than 50% of the requested size width and the supported size height should fill more than 70% of the requested size height. If the supported ad sizes doesn't satisfy the above conditions we send with size -1, -1. So in the example cases you provided:
* Requested ad size: [width: 729, height: 50], banner size of [width: 320, height: 50] is not expected as it doesn't fill 50% of the requested width hence got [width: -1, height: -1].
* Second case works fine.
* Requested ad size: [width: 600, height: 300], medium rectangle size of [width: 300, height: 250] doesn't fill more than 50% of the requested width hence got [width: -1, height: -1].

So please revert back to the code we have added i.e, `GADClosestValidSizeForAdSizes`.

File: adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m (hunk 2 above)
Comment (maninomula) [reply adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m]: Refer to the above comment.

File: adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m (hunk 1 above)
Comment (lcsabaignat) [reply adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m]: Thank you for the clarifications @maninomula!

**Comment by lcsabaignat** [comment]
@maninomula Re-added the size changes. Ready for review! And sorry for the delay.

**Pull request merged**